packages/block-library/src/embed/theme.scss

```diff
@@ -2,6 +2,6 @@
 	@include caption-style-theme();
 }
 
-:where(.wp-block-embed) {
+.wp-block-embed {
```

Comment:
Should the `.wp-block-embed figcaption` rule have lowered specificity?

Rather than adding `:root` to each individual rule, I thought it might be better to do something like this:
```scss
:root {
    :where(.wp-block-embed) {
        // ...
    }
}
```

This way everything is consistent.

I realize it's the same specificity as just a classname, but it makes it easier for anyone modifying the css to maintain the same specificity.

In the future if we ever need to implement layers, we can replace the `:root` with `@layer name`, or if we want to reduce specificity to zero we remove the `:root {}` wrapper.

Reply:
> Should the .wp-block-embed figcaption rule at the top of this file have lowered specificity?

I'd say so. A rough rule could be "anything that is configurable via global styles should be".

At a glance that mixin touches `color` and `font-size` rules and captions are now a configurable element in Global Styles.


> Rather than adding :root to each individual rule or removing :where, I thought it might be better to do something like this:

Sounds good to me. I'm still working through the impacts this PR will have on https://github.com/WordPress/gutenberg/pull/61032 and haven't gone through all the block styles yet.

My understanding is this PR has only tweaked what was changed in the original zero-specificity PRs which has proven to be somewhat incomplete. There was some discussion that we needed a complete audit of all the block styles once everything settled.

Reply:
One possible issue with the suggested approach to nesting `:root` and `:where` like that is it might encourage people to think they can still just nest the next selector within the `:where` when they really have to define the entire selector e.g.

```scss
:root {
    :where(.wp-block-button) {
        // ...
    }
    :where(.wp-block-button > .wp-block-button__link) {
        // ...
    }
}
```

Reply:
Yeah, it gets a bit much doesn't it. 😬